.ci/update-tekton-bundle.sh

```diff
@@ -0,0 +1,87 @@
+APPSTUDIO_TASKS_REPO=quay.io/repository/redhat-appstudio-qe/test-images
+PIPELINE_BUNDLE_IMG=quay.io/repository/redhat-appstudio-qe/test-images:pipeline-bundle-$JVM_BUILD_SERVICE_PR_SHA
+BUILD_DEFINITIONS_DIR="$WORKSPACE"/tmp/build-definitions
+
+function cloneBuildDefinitions() {
+    git clone https://$GITHUB_TOKEN@github.com/redhat-appstudio/build-definitions.git "$BUILD_DEFINITIONS_DIR"
+}
+
+function updateAnalyzerImage() {
+    yq e -i "(.[] \
+    | select(.path == \"/spec/steps/-\").value \
+    | select(.name == \"analyse-dependencies-java-sbom\").image) \
+    |= \"$JVM_BUILD_SERVICE_ANALYZER_IMAGE\"" \
+    $BUILD_DEFINITIONS_DIR/tasks/patches/add-sbom-and-push.yaml
+}
+
+function createDockerConfig() {
+    mkdir ~/.docker
+    echo $QUAY_TOKEN | base64 -d > ~/.docker/config.json
+}
+
+function createAndPushTaskBundle() {
+    if [[ $(uname) = Darwin ]]; then
+        CSPLIT_CMD="gcsplit"
+    else
+        CSPLIT_CMD="csplit"
+    fi
+
+    PART=1
+    COUNT=0
+    TASK_TEMP=$(mktemp -d)
+    PIPELINE_TEMP=$(mktemp -d)
+    oc kustomize $BUILD_DEFINITIONS_DIR/tasks | $CSPLIT_CMD -s -f $TASK_TEMP/task -b %02d.yaml - /^---$/ '{*}'
+    oc kustomize $BUILD_DEFINITIONS_DIR/pipelines/base > ${PIPELINE_TEMP}/base.yaml
+
+    ## Limit number of tasks in bundle
+    MAX=10
+    TASKS=
+    REF="$APPSTUDIO_TASKS_REPO:task-bundle-$JVM_BUILD_SERVICE_PR_SHA-$PART"
+    for TASK in $TASK_TEMP/task*.yaml; do
+        TASK_NAME=$(yq e ".metadata.name" $TASK)
+        if [ $COUNT -eq $MAX ]; then
+            echo Creating $REF
+            tkn bundle push $TASKS $REF
+            COUNT=0
+            PART=$((PART+1))
+            TASKS=
+        fi
+        REF="$APPSTUDIO_TASKS_REPO:task-bundle-$JVM_BUILD_SERVICE_PR_SHA-$PART"
+
+        # Replace appstudio-utils placeholder by newly build appstudio-utils image
+        #yq e -i ".spec.steps[] |= select(.image == \"appstudio-utils\").image=\"$APPSTUDIO_UTILS_IMG\"" $TASK
+
+        TASKS="$TASKS -f $TASK"
+        for file in $PIPELINE_TEMP/*.yaml; do
+            yq e -i "(.spec.tasks[].taskRef | select(.name == \"$TASK_NAME\")) |= {\"name\": \"$TASK_NAME\", \"bundle\":\"$REF\"}" $file
+            yq e -i "(.spec.finally[].taskRef | select(.name == \"$TASK_NAME\")) |= {\"name\": \"$TASK_NAME\", \"bundle\":\"$REF\"}" $file
+        done
+        COUNT=$((COUNT+1))
+    done
+    echo Creating $APPSTUDIO_TASKS_REPO:task-bundle-$JVM_BUILD_SERVICE_PR_SHA-$PART
+    tkn bundle push $TASKS $REF
+}
+
+function createAndPushPipelineBundles() {
+    echo Creating $PIPELINE_BUNDLE_IMG
+    tkn bundle push $PIPELINE_BUNDLE_IMG -f ${PIPELINE_TEMP}/base.yaml
+}
+
+function createConfigMap() {
+    echo Creating configmap to override default bundle to $PIPELINE_BUNDLE_IMG
+    oc apply -n build-templates -f - << CONFIGMAP
```

Comment:
this will be overridden by argocd

Reply:
That's a good point.
We could create this config map in a specific namespace that will be then used by the test, but since some tests are already using namespaces which names are random-generated (and I think jvm-build-service tests should do that too), we could leave the config map creation for the test itself (similarly like [here](https://github.com/redhat-appstudio/e2e-tests/blob/1069888eeed24803a3a904fbf0f94edfdf3cfc9f/tests/build/build.go#L489-L494)) - wdyt @gabemontero?

The bundle image value could be extracted from an env var. If the env var is empty, no config map will be created by the test.

Reply:
Seems like a good approach, I can remove this function here and just leave the env variable set. I

Reply:
I had a chat with @Michkov and we decided to patch the kustomization script in infra-deployments, so that the new build-bundle would be available by default to all namespaces.

Notice that, to do that, we needed to change the execution of this script to run before bootstraping the cluster. We are also changing the preview script in https://github.com/redhat-appstudio/infra-deployments/pull/526.